src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -1,654 +1,650 @@
-            }
-            currentPathTarget = pos;
-            return GUARD_SEARCH_TARGET;
-        }
-        else if (building.getTask().equals(AbstractBuildingGuards.Task.FOLLOW))
-        {
-            worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.following"));
+                worker.faceEntity(target, (float) TURN_AROUND, (float) TURN_AROUND);
+                worker.getLookHelper().setLookPositionWithEntity(target, (float) TURN_AROUND, (float) TURN_AROUND);
 
-            BlockPos pos = building.getPlayerToFollow();
-            if (pos == null || !building.getColony().isCoordInColony(world, pos))
-            {
-                if (pos != null)
+                worker.swingArm(EnumHand.MAIN_HAND);
+                worker.playSound(SoundEvents.ENTITY_PLAYER_ATTACK_SWEEP, (float) BASIC_VOLUME, (float) getRandomPitch());
+
+                double damageToBeDealt = 3;
+
+                if (worker.getHealth() <= DOUBLE_DAMAGE_THRESHOLD)
                 {
-                    Log.getLogger().info(BlockPosUtil.getDistance2D(pos, building.getColony().getCenter()));
+                    damageToBeDealt *= 2;
                 }
-                final EntityPlayer player = building.getPlayer();
-                if (player != null)
+
+                final ItemStack heldItem = worker.getHeldItem(EnumHand.MAIN_HAND);
+
+                if (ItemStackUtils.doesItemServeAsWeapon(heldItem))
                 {
-                    LanguageHandler.sendPlayerMessage(building.getPlayer(), "com.minecolonies.coremod.job.guard.switch");
+                    if(heldItem.getItem() instanceof ItemSword)
+                    {
+                        damageToBeDealt += ((ItemSword) heldItem.getItem()).getAttackDamage();
+                    }
+                    else
+                    {
+                        damageToBeDealt += TinkersWeaponHelper.getDamage(heldItem);
+                    }
+                    damageToBeDealt += EnchantmentHelper.getModifierForCreature(heldItem, target.getCreatureAttribute());
                 }
-                pos = building.getLocation();
-                building.setTask(AbstractBuildingGuards.Task.GUARD);
+
+                target.attackEntityFrom(new DamageSource(worker.getName()), (float) damageToBeDealt);
+                target.setRevengeTarget(worker);
+
+                worker.damageItemInHand(EnumHand.MAIN_HAND, 1);
             }
-            currentPathTarget = pos;
-            return GUARD_SEARCH_TARGET;
         }
-        worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.patrolling"));
-
-        currentPathTarget = getRandomBuilding();
-        return GUARD_SEARCH_TARGET;
+        return GUARD_ATTACK_PHYSICAL;
     }
 
-    /**
-     * Gets a random building from his colony.
-     *
-     * @return a random blockPos.
-     */
-    private BlockPos getRandomBuilding()
+    protected AIState attackRanged()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)